actionmetadata_test.go

```diff
@@ -122,3 +131,32 @@ func TestKnowledgeBase(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func testUrlpath(filePath string) *http.Response {
+	var index, count int = 0, 0
```

Comment:
This method needs to use the go-github client, not the http client directly. 

You can modularize and re-use code starting from this line:
https://github.com/step-security/secure-workflows/blob/1bd3f566c0c735e3039ae1d58d8763eb6a3f9a7b/pinactions.go#L49